kernel/src/vdso.rs

```diff
@@ -337,3 +339,51 @@ pub(crate) fn vdso_vmo() -> Option<Arc<Vmo>> {
     // We allow that VDSO does not exist
     VDSO.get().map(|vdso| vdso.vmo.clone())
 }
+
+#[cfg(target_arch = "x86_64")]
+pub const VDSO_VMO_LAYOUT: VdsoVmoLayout =
+    VdsoVmoLayout::new(0, PAGE_SIZE, 4 * PAGE_SIZE, PAGE_SIZE, 5 * PAGE_SIZE, 0x80);
+
+#[cfg(target_arch = "riscv64")]
+pub const VDSO_VMO_LAYOUT: VdsoVmoLayout =
+    VdsoVmoLayout::new(0, PAGE_SIZE, 2 * PAGE_SIZE, PAGE_SIZE, 3 * PAGE_SIZE, 0);
+
+pub struct VdsoVmoLayout {
+    pub data_segment_offset: usize,
+    pub data_segment_size: usize,
+    pub text_segment_offset: usize,
+    pub text_segment_size: usize,
+    pub size: usize,
+
+    pub data_offset: usize,
+}
+
+impl VdsoVmoLayout {
+    pub const fn new(
+        data_segment_offset: usize,
+        data_segment_size: usize,
+        text_segment_offset: usize,
+        text_segment_size: usize,
+        size: usize,
+        data_offset: usize,
+    ) -> Self {
+        Self {
+            data_segment_offset,
+            data_segment_size,
+            text_segment_offset,
+            text_segment_size,
+            size,
+            data_offset,
+        }
+    }
+}
+
+const_assert!(VDSO_VMO_LAYOUT.data_segment_offset % PAGE_SIZE == 0);
+const_assert!(VDSO_VMO_LAYOUT.data_segment_size % PAGE_SIZE == 0);
+const_assert!(VDSO_VMO_LAYOUT.text_segment_offset % PAGE_SIZE == 0);
+const_assert!(VDSO_VMO_LAYOUT.text_segment_size % PAGE_SIZE == 0);
+const_assert!(VDSO_VMO_LAYOUT.size % PAGE_SIZE == 0);
+const_assert!(
+    VDSO_VMO_LAYOUT.data_offset + size_of::<VdsoData>()
+        <= VDSO_VMO_LAYOUT.data_segment_offset + VDSO_VMO_LAYOUT.data_segment_size
```

Comment:
This checks the end of the vDSO data range but does not check the start?
```rust
// Ensure that the vDSO data at `DATA_OFFSET` is in the data segment.
const_assert!(VdsoLayout::DATA_OFFSET >= VdsoLayout::DATA_SEGMENT_OFFSET);
const_assert!(
    VdsoLayout::DATA_OFFSET + size_of::<VdsoData>()
        <= VdsoLayout::DATA_SEGMENT_OFFSET + VdsoLayout::DATA_SEGMENT_SIZE
);
```

Also I think we should keep the comment.